Sources/ComposedUI/Common/ChangesReducer.swift

```diff
@@ -375,13 +375,13 @@ internal struct ChangesReducer: CustomReflectable {
             indexPath.section == section
         }
 
-        let itemsRemoved = changeset.elementsRemoved.filter(isIncluded(indexPath:)).map(\.item)
-        let itemsInserted = changeset.elementsInserted.filter(isIncluded(indexPath:)).map(\.item)
+        let itemsRemoved = changeset.elementsRemoved.filter(isIncluded(indexPath:))
+        let itemsInserted = changeset.elementsInserted.filter(isIncluded(indexPath:))
 
         let availableSpaces = (0..<Int.max)
             .lazy
-            .filter { !itemsRemoved.contains($0) }
-        let item = item - itemsInserted.filter({ $0 < item }).count
+            .filter { !itemsRemoved.contains(IndexPath(item: $0, section: section)) }
```

Comment:
@JosephDuffy just want to check with you that the behavior will as expected after we use `IndexPath` to compare.
I think we have used `isIncluded` to filter out the same section so we can init a `IndexPath` here with section from function parameter.